scripts/linters/js_ts_linter.py

```diff
@@ -843,21 +843,25 @@ def _check_constants_declaration(self):
             # *.constants.ajs.ts file are in sync.
             if filepath.endswith('.constants.ts') and (
                     is_corresponding_angularjs_filepath):
-                # Ignore if file contains only type definitions for
-                # constants.
-                for node in parsed_nodes:
-                    if 'declarations' in node.keys():
+                if filepath.endswith('app.constants.ts'):
```

Comment:
Maybe add a comment here with some explanation of why we are treating the two cases differently.

Also I feel like you can reduce the branching?

Lastly -- why do you have "except Exception: continue"? In what cases will there be an exception, and can we be more explicit about this? It basically sounds like "ignore errors" and I think that just catching a general Exception is too broad. If the issue is something like "expression is None" then just check for that instead.

Reply:
> Maybe add a comment here with some explanation of why we are treating the two cases differently.
done

> Also I feel like you can reduce the branching?
done

> Lastly -- why do you have "except Exception: continue"? In what cases will there be an exception, and can we be more explicit about this? It basically sounds like "ignore errors" and I think that just catching a general Exception is too broad. If the issue is something like "expression is None" then just check for that instead.

This is needed because the `parsed_nodes` look something like https://pastebin.com/mfL4m8DS in `app.constants.ts`. Now we need a node that exactly has `node.expression.right.arguments[1].properties` properties. There might be some nodes that have only `expression` or only `expression.right`. Hence, in this case, we would have to do something like

```python
if node.expression:
  if node.expression.right:
    if node.expression.arguments:
      # And so on.....
```

So, I think `continue` can be used here. Also, this was the way it was done before I modified the expression.

Reply:
Thanks for the detailed explanation. In that case, could you use KeyError or AttributeError then (or something that's not so broad as Exception)? (Maybe add an explanation too, in the code.)

Reply:
done!